#! /bin/sh /usr/share/dpatch/dpatch-run
## 940_test_thread_mutex_sync_shorter.dpatch by Lucas Nussbaum <lucas@lucas-nussbaum.net>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: test_mutex_synchronize takes a very long time on slow arches,
## DP: especially hppa (since it doesn't support NPTL).
## DP: The patch makes the test shorter by creating less concurrent
## DP: threads. Closes: #514696.

@DPATCH@
Index: b/test/ruby/test_thread.rb
===================================================================
--- a/test/ruby/test_thread.rb	2010-07-20 21:45:45.000000000 +0900
+++ b/test/ruby/test_thread.rb	2010-07-20 21:46:16.000000000 +0900
@@ -30,7 +30,7 @@
   def test_mutex_synchronize
     m = Mutex.new
     r = 0
-    max = 100
+    max = 10
     (1..max).map{
       Thread.new{
         i=0
